base_location_geonames_import/wizard/geonames_import.py

```diff
@@ -103,20 +104,33 @@ def get_and_parse_csv(self, country):
         url = config_url % country_code
         logger.info("Starting to download %s" % url)
         res_request = requests.get(url, timeout=15)
-        if res_request.status_code != requests.codes.ok:
-            # pylint: disable=translation-positional-used - Don't want to re-translate
-            raise UserError(
-                _("Got an error %d when trying to download the file %s.")
-                % (res_request.status_code, url)
-            )
 
-        f_geonames = zipfile.ZipFile(io.BytesIO(res_request.content))
-        tempdir = tempfile.mkdtemp(prefix="odoo")
-        f_geonames.extract("%s.txt" % country_code, tempdir)
+        if res_request.status_code == requests.codes.ok:
+            f_geonames = zipfile.ZipFile(io.BytesIO(res_request.content))
+            tempdir = tempfile.mkdtemp(prefix="odoo")
+            f_geonames.extract("%s.txt" % country_code, tempdir)
 
-        data_file = open(
-            os.path.join(tempdir, "%s.txt" % country_code), "r", encoding="utf-8"
-        )
+            f_route = os.path.join(tempdir, "%s.txt" % country_code)
+        else:
+            logger.warning(
+                _(
+                    f"Got an error {res_request.status_code} "
+                    f"when trying to download the file {url}."
+                    " Searching for the file locally..."
+                )
+            )
+            addon_dir = get_module_path("base_location_geonames_import")
+            f_route = os.path.join(addon_dir, f"data/zips/{country_code}.txt")
+            if not os.path.exists(f_route):
+                # pylint: disable=translation-positional-used - Don't want to re-translate
```

Comment:
Please don't

Reply:
What do you mean?

Reply:
Don't disable pylint errors, but solve them.